codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/DocumentShapeDeserVisitor.java

```diff
@@ -414,7 +415,12 @@ protected final void generateDeserFunction(
         GoWriter writer = context.getWriter();
 
         Symbol symbol = symbolProvider.toSymbol(shape);
-        String functionName = ProtocolGenerator.getDocumentDeserializerFunctionName(shape, context.getProtocolName());
+
+        String functionName = shape.hasTrait(SyntheticClone.class)
+                ? ProtocolGenerator.getOperationDocumentDeserFuncName(
+                        shape, context.getProtocolName())
+                : ProtocolGenerator.getDocumentDeserializerFunctionName(
+                        shape, context.getService(), context.getProtocolName());
```

Comment:
nit: Should this be a helper utility somewhere as this logic is in multiple spots?

Reply:
I will just put this logic back in getDocumentDeserializerFunctionName function and centralize this there. I pulled it out since it seemed out of place,but thinking again, I think it's ok to have it there.